tsconfig-dist.json

```diff
@@ -0,0 +1,36 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "module": "es2015",
```

Comment:
@smnbbrv as far as I get it, to your library users it does not really matter which tool you use in order to bundle your library (same story as TypeScript version). You could have used rollup, or gulp, or manually listed and imported single files (many libraries do that). At the end of the day, what matters should be the output files they get, and more specific things like:
* which module system you use to group classes (e.g. `es2015`, `commonjs`), so that users can import those
* to have a public `NgModule` defined, to make it work with angular 2 post beta/rc
* to have `metadata.json` files, to make it work with angular 2 AOT build
* to have `d.ts` definition files, to make it work with users in TypeScript projects

So I'm not sure I get what is the issue with webpack2 Vs webpack 1. Re. module system, I guess you can move to `commonjs` if you think that could be an issue for users.